### ISSUE TITLE
Refactor theme implementation 

### ISSUE THREAD

**Issue by fm-sys** [cgeo/cgeo#9373]
Posted in https://github.com/cgeo/cgeo/pull/9291#issuecomment-723480389 
> I found no way [...] for setting the background according to our theme settings at this early stage of loading program

Our current theme implementation is quite outdated. Maybe we should consider refactoring it for making use of optional automatic theme switches according to the device settings, easy use of different colored svg icons according to the theme and of course using the right background also for the startup animation.

I'm in no way firm with this topic, so maybe I mixed up some things, but AFAIK our theme implementation exists since nearly 10 years where Android itself wasn't really made for supporting multiple themes at all... 

The current way, how multiple themes should be implemented is described here: https://developer.android.com/guide/topics/ui/look-and-feel/darktheme

**Comment by moving-bits** [comment]
Refactoring the whole theming would be nice indeed. Currently it's a crude/grown mix along several Android versions and theming concepts, which makes it hard to have a consistent look and feel.

Since we have dropped support for Android 4 it may be a good starting point for a refactoring. We could use a new base theme which supports all colors and other parameters we need for c:geo, and build all other styles/themes on top of it.

**Comment by moving-bits** [comment]
I allowed myself to change the issue title to a more generalized "Refactor theme implementation", as it's not alone the light theme implementation that needs to be refactored. Hope that's ok for you.

**Comment by moving-bits** [comment]
Just as a quick test I did a "create new project" today in AS to see what an empty app would look like today. I set min API level to 21, and it generated something like this:

themes.xml is based on "Theme.MaterialComponents.DayNight.DarkActionBar":
```
<resources xmlns:tools="http://schemas.android.com/tools">
    <!-- Base application theme. -->
    <style name="Theme.Basicstyling" parent="Theme.MaterialComponents.DayNight.DarkActionBar">
        <!-- Primary brand color. -->
        <item name="colorPrimary">@color/purple_500</item>
        <item name="colorPrimaryVariant">@color/purple_700</item>
        <item name="colorOnPrimary">@color/white</item>
        <!-- Secondary brand color. -->
        <item name="colorSecondary">@color/teal_200</item>
        <item name="colorSecondaryVariant">@color/teal_700</item>
        <item name="colorOnSecondary">@color/black</item>
        <!-- Status bar color. -->
        <item name="android:statusBarColor" tools:targetApi="l">?attr/colorPrimaryVariant</item>
        <!-- Customize your theme here. -->
    </style>
</resources>
```

MainActivity is based on `AppCompatActivity`:
```
public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }
}
```

And the activity's layout is based on `ConstraintLayout`:
```
<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout
    xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="match_parent"
    tools:context=".MainActivity">
    <TextView
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:text="Hello World!"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintLeft_toLeftOf="parent"
        app:layout_constraintRight_toRightOf="parent"
        app:layout_constraintTop_toTopOf="parent" />
</androidx.constraintlayout.widget.ConstraintLayout>
```

Maybe that's something to start with for us...

**Comment by fm-sys** [comment]
From https://github.com/cgeo/cgeo/pull/10630#discussion_r633088795 (@moving-bits):
> With regards to new base layout: I've also started with a new base layout, including some nice method for switching between dark and light mode, but was blocked by the fact, that our preferences are based on a different type of activities, not being compatible to AppCompatActivity, which would require us to maintain two sets of themes/styles. That's why I opened #10579.

How far have you come already? 

I myself would call myself a bit familiar with it. Either I could take over something, or give you a little help. Here are two commits of mine, where I implemented theme switching according to current standards:

1. Implement theme switch (parent theme: "Theme.MaterialComponents.DayNight" - one single theme for light and dark):
https://github.com/fm-sys/snapdrop-android/commit/80700a9fa593357f72cac2d350b4a9c3ff5d8ae3

1. Dark mode following system default feature (parent theme: "Theme.MaterialComponents.DayNight" as well):
https://github.com/stefan-niedermann/nextcloud-deck/pull/755/commits/3bd8b5ed2ed26fb57f8f14e9919947d61fdbbee4

So if we refactor our themes, there should really be **only one single theme** for everything. The colors should be structured in two different resource folders: `values/colors.xml` and `values-night/colors.xml`. 

e.g. the theme contains `<item name="colorPrimary">@color/primary</item>` - the primary color will be chosen according to the current theme setting:
- `values/colors.xml`: all colors for light theme
- `values-night/colors.xml`: all colors for dark theme

**Comment by moving-bits** [comment]
@fm-sys 
Thanks for sharing theses ideas. Looks like I've followed a very similar approach last week... :-)
For easier discussion and a joint approach I've uploaded what I've put together so far, see #10643 - just as a reference and for discussing ideas.

**Comment by eddiemuc** [comment]
Hm from what I see I guess it is best if I step back for now and wait for you guys to figure out best way to work with themes/styles in the future :-)
No worries, just at some point give me a hint as to what base theme to use.

**Comment by fm-sys** [comment]
@moving-bits #10643 looks already very promising and it's probably the best if you continue it.

For adding a "choose automatically" mode to the theme setting, I would really like to provide the code (as I have already done it twice). But this is only possible as soon as your PR is merged. Just wanna say that you don't have to step into this area while refactoring the base theme...

**Comment by fm-sys** [comment]
Meanwhile done. For the remaining things we have follow up issues...
